src/eduid_userdb/actions/db.py

```diff
@@ -51,6 +52,7 @@ def __init__(self, db_uri, db_name='eduid_actions', collection='actions'):
         super(ActionDB, self).__init__(db_uri, db_name, collection)
 
         self._cache = {}
+        self._left_in_cache = {}
```

Comment:
Why have _left_in_cache instead of just checking len(self._cache[cachekey])?

To me, reading this diff, it appears to just introduce possible bugs with out-of-sync info in _cache and _left_in_cache.